src/jinja2/filters.py

```diff
@@ -765,7 +788,7 @@ def do_int(value, default=0, base=10):
             return default
 
 
-def do_float(value, default=0.0):
+def do_float(value: Union[int, float, str], default=0.0) -> float:
```

Comment:
I'm not sure if `Any` would be more appropriate here.
```suggestion
def do_float(value: Any, default=0.0) -> float:
```

Reply:
I did it in this manner because you would not run a do_float on a list or dict and if we use Any that might suggest such a thing could/would get a valid return statement.

Reply:
Yeah that's true. I think a type annotation for default (`float`) would also help :D

Reply:
In this case I do not think we would want someone using `do_float` on a list or dict so I think Any would be a little to vague.